src/utils/stores.ts

```diff
@@ -19,7 +19,13 @@ import { fetchTokenDecimals, fetchTokenName, fetchTokenSymbol } from "../utils/t
 import { ADDRESS_ZERO, BD_ZERO, BI_ZERO } from "./numbers"
 
 export function getBlockNumberLogIndex(event: ethereum.Event): BigInt {
-    return event.block.number.times(BigInt.fromI32(1000)).plus(event.logIndex)
+    // NOTE:
+    // after bedrock upgrade, the current gasLimit per block is 25M
+    // the maximum amount of tx per block is 25M / 21K ~= 1190.48
+    // (the minimum gas usage of tx is 21K)
+    // BlockNumberLogIndex = block.number * 10_000_000 + txLogIndex * 1_000 + logIndex
+    const index = event.transactionLogIndex.times(BigInt.fromI32(1_000)).plus(event.logIndex)
+    return event.block.number.times(BigInt.fromI32(10_000_000)).plus(index)
```

Comment:
According to https://docs.ethers.org/v5/api/providers/types/#providers-Log, `logIndex` means the index of this log across all logs in the entire block, so I think we don't need `transactionLogIndex`.

However, since one block could contain multiple transactions now, and gas limit is 25,000,000 per block. The `1000` gap is not enough. `10_000_000` should be ok.

Reply:
Just notice there is another `transactionLogIndex `!!

Reply:
@vinta just query our subgraph and confirmed that only `logIndex` is enough.